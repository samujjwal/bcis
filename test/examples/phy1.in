# Calculate Factorial 
# This program is testing Looping fucntion 
# Test Program written by Changlong Jiang cj2214@columbia.edu
# Date 12/15/2007

# use For Loops
set y=1
for x from 1 to 10 step 1 do
set y = y*x
done
print("use For Loop")
print(y)	

# use While Loops
set y=1
set z=1
while y <= 10 do
set z = z * y
set y = y+1
done
print("use While Loop")
print(z)

#use IF 
print("use While Loop and If")
set y=1
set z=1
while y <= 10 do
set z = z * y
  if y>9 then
    print("y=",y," ","greater than 9, stop")
    return y;
  elsif y>6 then
    print("y=",y," ","greater than 6, continue")
    print("result=",z)
  else
    print("y=",y," ","less and equal than 6,continue")
    print("result=",z) 
  done
set y = y+1
done



