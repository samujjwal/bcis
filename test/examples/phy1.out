use For Loop
3628800.0
use While Loop
3628800.0
use While Loop and If
y=1.0 less and equal than 6,continue
result=1.0
y=2.0 less and equal than 6,continue
result=2.0
y=3.0 less and equal than 6,continue
result=6.0
y=4.0 less and equal than 6,continue
result=24.0
y=5.0 less and equal than 6,continue
result=120.0
y=6.0 less and equal than 6,continue
result=720.0
y=7.0 greater than 6, continue
result=5040.0
y=8.0 greater than 6, continue
result=40320.0
y=9.0 greater than 6, continue
result=362880.0
y=10.0 greater than 9, stop
